### ISSUE TITLE
Rural Zinc Chameleon - No pauseable modifier in the limit order contract can have unpredictable results when upgrading the protocol

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2023-12-flatmoney-judging#34]
Rural Zinc Chameleon

medium

# No pauseable modifier in the limit order contract can have unpredictable results when upgrading the protocol

## Summary

No pauseable modifier in the limit order contract can have unpredictable results when upgrading the protocol

## Vulnerability Detail

Observe the following code 

https://github.com/sherlock-audit/2023-12-flatmoney/blob/bba4f077a64f43fbd565f8983388d0e985cb85db/flatcoin-v1/src/DelayedOrder.sol#L67

In the delayed Order contract announce functions all have a pauseable modifier, to stop users from calling the functions, during times of high volatility or in the midst of a protocol upgrade. However, the announcelimitorder function in the limit order contract do not have this modifier 

```solidity
  function announceLimitOrder(uint256 tokenId, uint256 priceLowerThreshold, uint256 priceUpperThreshold) external {
        uint64 executableAtTime = _prepareAnnouncementOrder();
        address positionOwner = _checkPositionOwner(tokenId);
        _checkThresholds(priceLowerThreshold, priceUpperThreshold);
        uint256 tradeFee = ILeverageModule(vault.moduleAddress(FlatcoinModuleKeys._LEVERAGE_MODULE_KEY)).getTradeFee(
            vault.getPosition(tokenId).additionalSize
        );
```

The absence of this modifier will allow users to undermine the owner, bypass the pauseable modifiers in the delayed order contracts and close positions. This can result in unpredictable behavior if the protocol is in the midst of an upgrade to its business logic or in times of high volatility 

https://github.com/sherlock-audit/2023-12-flatmoney/blob/bba4f077a64f43fbd565f8983388d0e985cb85db/flatcoin-v1/src/LimitOrder.sol#L58-L64

## Impact

Users can bypass pauseable modifiers which can be potentially devasting if in the midst of an upgrade or times of high volitility 

## Code Snippet
https://github.com/sherlock-audit/2023-12-flatmoney/blob/bba4f077a64f43fbd565f8983388d0e985cb85db/flatcoin-v1/src/DelayedOrder.sol#L67

## Tool used

Manual Review

## Recommendation

 Add a pauseable modifier to the limit order function just like in the delayed order contract

**Comment by nevillehuang** [comment]
Invalid, speculation on future emergencies requiring a pause mechanism. Additionally no funds are loss as orders would still have to be put up and excecuted

**Comment by RealLTDingZhen** [comment]
Hi @rcstanciu Sorry to disturb, could this issue(and its dup) be seen as breaking the core functionality of the protocol?

**Comment by RealLTDingZhen** [comment]
Hi @rashtrakoff would you like to take a look at this and its dup?

**Comment by rashtrakoff** [comment]
Yeah I would say this is invalid. LimitOrders are not one of the core contracts. Core contracts being LeverageModule and StableModule which if not paused can be an issue. So if these two are paused then LimitOrders and even DelayedOrders execution will fail.

We intend to put whenNotPaused regardless for consistency between all modules.
